### PR TITLE
remove the dependency on gfortran during the MKL compilation process

### PR DESCRIPTION
**Pull request by inabao**

### CODE REVIEW

File: extern/mkl/mkl.cmake [hunk 1]
```diff
@@ -58,7 +58,7 @@ if (CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86_64" AND ENABLE_INTEL_MKL)
     endforeach()
     message ("enable intel-mkl as blas backend")
 else ()
-    set(BLAS_LIBRARIES libopenblas.a)
+    set(BLAS_LIBRARIES libopenblas.a gfortran)
```
Comment (ZhitaoShen) [inline extern/mkl/mkl.cmake:61]: openblas depends on gfortran library.

File: tests/test_hnsw_new.cpp [hunk 2]
```diff
@@ -169,7 +169,7 @@ TEST_CASE_METHOD(fixtures::TestIndex, "HNSW Factory Test With Exceptions", "[ft]
     SECTION("Invalid hnsw param ef_construction") {
         auto ef_construction = GENERATE(-1, 0, 100000, 31);
         // TODO(LHT): test for float param
-        auto param_temp =
+        constexpr const char* param_temp =
```
Comment (ZhitaoShen) [inline tests/test_hnsw_new.cpp]: fix compiler warning in clang17

**Review by ZhitaoShen — APPROVED** [review]
LGTM

**Review by jiaweizone — APPROVED** [review]
LGTM

**Pull request merged**